File: booking_hotel_backend/src/main/java/com/group/hotel/service/RoomService.java
package com.group.hotel.service;

import com.group.hotel.dto.request.room.RoomCreateRequest;
import com.group.hotel.dto.request.room.RoomSearchRequest;
import com.group.hotel.dto.request.room.RoomUpdateRequest;
import com.group.hotel.common.response.PageResponse;
import com.group.hotel.dto.response.room.RoomDetailResponse;
import com.group.hotel.dto.response.room.RoomResponse;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface RoomService {
    List<String> getStatuses();
    List<String> getRoomTypes();
    PageResponse<RoomResponse> getAll(RoomSearchRequest roomSearchRequest, Pageable pageable);
    RoomDetailResponse getDetail(Long id);
    RoomResponse create(RoomCreateRequest roomCreateRequest);
    RoomResponse update(Long id, RoomUpdateRequest roomUpdateRequest);
    void delete(Long id);
    RoomResponse restore(Long id);
    RoomDetailResponse updateFurnitures(Long id, List<Long> furnitureIds);
    String uploadImage(Long id, MultipartFile file);
}

File: booking_hotel_frontend/src/services/roomService.js
import axiosInstance from "./axiosInstance";

const roomService = {
    getAll: async (params) => {
        const response = await axiosInstance.get("/rooms", { params });
        return response.data;
    },
    getDetail: async (id) => {
        const response = await axiosInstance.get(`/rooms/${id}`);
        return response.data;
    },
    getStatuses: async () => {
        const response = await axiosInstance.get("/rooms/statuses");
        return response.data;
    },
    getRoomTypes: async () => {
        const response = await axiosInstance.get("/rooms/room-type");
        return response.data;
    },
    create: async (data) => {
        const response = await axiosInstance.post("/rooms", data);
        return response.data;
    },
    update: async (id, data) => {
        const response = await axiosInstance.put(`/rooms/${id}`, data);
        return response.data;
    },
    delete: async (id) => {
        const response = await axiosInstance.delete(`/rooms/${id}`);
        return response.data;
    },
    restore: async (id) => {
        const response = await axiosInstance.put(`/rooms/${id}/restore`);
        return response.data;
    },
    updateFurnitures: async (id, furnitureIds) => {
        const response = await axiosInstance.put(`/rooms/${id}/furnitures`, furnitureIds);
        return response.data;
    },
    uploadImage: async (id, file) => {
        const formData = new FormData();
        formData.append("file", file);
        const response = await axiosInstance.post(`/rooms/${id}/image`, formData, {
            headers: { "Content-Type": "multipart/form-data" },
        });
        return response.data;
    },
};

export default roomService;
